libkbfs/block_journal.go

```diff
@@ -762,28 +718,33 @@ func (j *blockJournal) removeFlushedEntry(ctx context.Context,
 		return err
 	}
 
-	// TODO: This is a hack to work around KBFS-1439. Figure out a
-	// better way to update j.refs to reflect the removed entry.
-	refs, err := j.readJournal(ctx)
-	if err != nil {
-		return err
+	for id, idContexts := range entry.Contexts {
+		refs := j.refs[id]
+		if len(refs) == 0 {
+			continue
+		}
+		for _, context := range idContexts {
+			err := refs.remove(context, earliestOrdinal)
```

Comment:
You could have one op (tag=1) that puts a ref, and then another (tag=2) that archives the same ref. Then when flushing the first op, the tags don't match. I guess that, and addref + archive are currently the only cases, right?

In the future we might put + remove + 2nd put, so I wanted to be robust against that, and other possibly data format changes, e.g. keeping around a tombstone for removed refs instead of removing it from the map.